### PR TITLE
Add weight feature to GeoIP backend

### PR DESCRIPTION
**Pull request by cmouse**
This allows administrator to define weights on records so that they are returned approximately weight% times on queries.

### CODE REVIEW

File: docs/markdown/authoritative/backend-geoip.md [hunk 1]
```diff
@@ -71,13 +71,20 @@ domains:
...
       - mx: 10 mx.example.com
     fin.eu.service.geo.example.com:
       - a: 198.51.100.221
       - txt: hello world
       - aaaa: 2001:DB8::12:34DE:3
+    swe.eu.service.geo.example.com
+      - a:
+           content: 1.2.3.4
+           weight: 50
+      - a: 2.3.4.5
```
Comment (ahupowerdns) [inline docs/markdown/authoritative/backend-geoip.md]: what is the weight of 2.3.4.5?

**Comment by Habbie** [comment]
(Please use documentation IP space like 192.0.2.0/24)

File: modules/geoipbackend/geoipbackend.cc [hunk 2]
```diff
@@ -123,9 +128,24 @@ void GeoIPBackend::initialize() {
...
+             string attr = iter->first.as<string>();
+             if (attr == "content") {
+	       string content = iter->second.as<string>();
+               rr.content = content;
+             } else if (attr == "weight") {
+               rr.weight = iter->second.as<int>();
+             } else if (attr == "ttl") {
+               rr.ttl = iter->second.as<int>();
+             } else {
+               L<<Logger::Warning<<"Unsupported record attribute " << attr << " for " << rr.qname << endl;
```
Comment (ahupowerdns) [inline modules/geoipbackend/geoipbackend.cc]: just kill it at this point. Will only lead to silently wrong config files.

File: modules/geoipbackend/geoipbackend.cc [hunk 3]
```diff
@@ -208,6 +230,39 @@ void GeoIPBackend::initialize() {
...
 
+    // finally fix weights
+    for(auto &item: dom.records) {
+      int weight=-1;
+      int rem, count;
+      rem = count = 0;
+      // first we look for used weight
+      for(const auto &rr: item.second) {
+        if (rr.weight > -1) {
+          if (weight==-1)
```
Comment (ahupowerdns) [inline modules/geoipbackend/geoipbackend.cc]: > -1 means it won't be -1

File: modules/geoipbackend/geoipbackend.cc (hunk 2 above)
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: well, it does log it. =)

**Comment by cmouse** [comment]
Rewrote to use weight "as usual". Please review again, sorry for rebase.

**Pull request merged**